autonomous/autonomous.py

```diff
@@ -28,98 +48,85 @@ class AutoBase(AutonomousStateMachine):
 
-    waypoints: List[Pose2d]
+    waypoints: List[Waypoint]
+    waypoints_poses: List[Pose2d]
+
+    max_speed = 2.5
+    max_accel = 1.5
 
     def __init__(self):
         super().__init__()
+        self.waypoints_poses = [w.pose for w in self.waypoints]
         # applies to the linear speed, not turning
-        self.linear_constraints = TrapezoidProfile.Constraints(1, 1)
+        self.linear_constraints = TrapezoidProfile.Constraints(
+            self.max_speed, self.max_accel
+        )
 
-        self.drive_rotation_constrants = trajectory.TrapezoidProfileRadians.Constraints(
-            2, 2
+        self.drive_rotation_constraints = (
+            trajectory.TrapezoidProfileRadians.Constraints(2, 2)
         )
 
         rotation_controller = controller.ProfiledPIDControllerRadians(
-            2, 0, 0, self.drive_rotation_constrants
+            2, 0, 0, self.drive_rotation_constraints
         )
         rotation_controller.enableContinuousInput(-math.pi, math.pi)
         self.drive_controller = controller.HolonomicDriveController(
-            controller.PIDController(0.5, 0, 0),
-            controller.PIDController(0.5, 0, 0),
+            controller.PIDController(1, 0, 0.2),
+            controller.PIDController(1, 0, 0.2),
             rotation_controller,
         )
 
         # all in meters along straight line path
-        self.total_length = trajectory_generator.total_length(self.waypoints)
+        self.total_length = trajectory_generator.total_length(self.waypoints_poses)
         # how far around the current position is used to smooth the path
         self.look_around = 0.3
+        self.cur_waypoint = 0
 
-        self.last_pose = self.waypoints[0]
+        self.last_pose = self.waypoints[0].pose
         self.trap_profile_start_time = 0
 
         wpilib.SmartDashboard.putNumber("auto_vel", 0.0)
 
     def setup(self):
-        field_bballs = self.field.getObject("bballs")
-        field_rballs = self.field.getObject("rballs")
         field_goal = self.field.getObject("goal")
-        field_bballs.setPoses(
-            [trajectory_generator.goal_to_field(Pose2d(*b, 0)) for b in blue_balls]
-        )
-        field_rballs.setPoses(
-            [trajectory_generator.goal_to_field(Pose2d(*b, 0)) for b in red_balls]
-        )
         field_goal.setPose(trajectory_generator.goal_to_field(Pose2d(0, 0, 0)))
 
     def on_enable(self):
-        self.target_estimator.set_pose(self.waypoints[0])
+        self.target_estimator.set_pose(self.waypoints[0].pose)
 
-        self.last_pose = self.waypoints[0]
+        self.last_pose = self.waypoints[0].pose
         # generates initial velocity profile
-        self.trap_profile, self.stop_point = self.generate_trap_profile(
-            len(self.waypoints), TrapezoidProfile.State(0, 0)
-        )
+        self.cur_waypoint = 0
+        self.trap_profile = self._generate_trap_profile(TrapezoidProfile.State(0, 0))
         return super().on_enable()
```

Comment:
nit: I know this isn't new code, but `on_enable` should return None, right?
```suggestion
        super().on_enable()
```